rockcraft/parts.py

```diff
@@ -58,12 +58,14 @@ def __init__(
         self,
         all_parts: Dict[str, Any],
         *,
+        project_name: str,
```

Comment:
was this required to populate CRAFT_PROJECT_NAME?

Reply:
yes. I could have used the `project_vars` dict for this, but `craft_parts.LifecycleManager` has a `project_name` constructor parameter which seemed like the more "correct" way to do this